Sources/NimiqClient/NimiqClient.swift

```diff
@@ -0,0 +1,1082 @@
+        var params:[Any] = [constant]
+        if value != nil {
+            params.append(value!)
+        }
+        return try call(method: "constant", params: params)
+    }
+
+    /// Creates a new account and stores its private key in the client store.
+    /// - Returns: Information on the wallet that was created using the command.
+    public func createAccount() throws -> Wallet? {
+        return try call(method: "createAccount", params: [])
+    }
+
+    /// Creates and signs a transaction without sending it. The transaction can then be send via `sendRawTransaction()` without accidentally replaying it.
+    /// - Parameter transaction: The transaction object.
+    /// - Returns: Hex-encoded transaction.
+    public func createRawTransaction(_ transaction: OutgoingTransaction) throws -> String? {
+        let params:[String:Any?] = [
+            "from": transaction.from,
+            "fromType": transaction.fromType?.rawValue,
+            "to": transaction.to,
+            "toType": transaction.toType?.rawValue,
+            "value": transaction.value,
+            "fee": transaction.fee,
+            "data": transaction.data
+        ]
+        return try call(method: "createRawTransaction", params: [params])
+    }
+
+    /// Returns details for the account of given address.
+    /// - Parameter address: Address to get account details.
+    /// - Returns: Details about the account. Returns the default empty basic account for non-existing accounts.
+    public func getAccount(address: Address) throws -> Any? {
+        let result: RawAccount = try call(method: "getAccount", params: [address])!
+        return result.value
+    }
+
+    /// Returns the balance of the account of given address.
+    /// - Parameter address: Address to check for balance.
+    /// - Returns: The current balance at the specified address (in smalest unit).
+    public func getBalance(address: Address) throws -> Int? {
+        return try call(method: "getBalance", params: [address])
+    }
+
+    /// Returns information about a block by hash.
+    /// - Parameter hash: Hash of the block to gather information on.
+    /// - Parameter fullTransactions: If `true` it returns the full transaction objects, if `false` only the hashes of the transactions.
+    /// - Returns: A block object or `nil` when no block was found.
+    public func getBlockByHash(_ hash: Hash, fullTransactions: Bool = false) throws -> Block? {
+        return try call(method: "getBlockByHash", params: [hash, fullTransactions])
+    }
+
+    /// Returns information about a block by block number.
+    /// - Parameter height: The height of the block to gather information on.
+    /// - Parameter fullTransactions: If `true` it returns the full transaction objects, if `false` only the hashes of the transactions.
+    /// - Returns: A block object or `nil` when no block was found.
+    public func getBlockByNumber(height: Int, fullTransactions: Bool = false) throws -> Block? {
```

Comment:
nitpick (non-blocking): this could be refactored to have "fluent design".

Last one I promise 😄 
Possible refactor to `public func block(atHeight height: Int, withTransactions fullTransactions)` resulting in this usage: `block(atHeight: 1000, withTransactions: true)`

Reply:
BTW, about the Swift naming conventions (and somewhat off-topic), I'm not sure why they dropped using `YES` and `NO` instead or `true` and `false`. Using YES and NO in the code for boolean values seems to me more "fluent" 🤔

Reply:
This is now `getBlock(atHeight height:, withTransactions:)`